File: 0002-add-two-numbers/0002-add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode * res = new ListNode(-1);
        ListNode * tail = res;

        int val1 = 0 , val2 = 0 , carry = 0 ;

        while(l1 || l2 || carry){
            val1 = (l1 ? l1->val : 0);
            val2 = (l2 ? l2->val : 0);

            int total = val1 + val2 + carry;

            carry = total > 9 ? 1 : 0;

            tail->next = new ListNode(total % 10);
            tail = tail->next;

            l1 = l1 ? l1->next : l1;
            l2 = l2 ? l2->next : l2; 
        }

        return res->next;
    }
};

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        val1 = 0 
        val2 = 0 
        carry = 0
        res = ListNode(-1)
        tail = res


        while l1 or l2 or carry:
            val1 = l1.val if l1 else 0
            val2 = l2.val if l2 else 0

            total = val1 + val2 + carry

            carry  = 1 if total > 9 else  0

            tail.next = ListNode(total % 10)
            tail = tail.next

            l1 = l1.next if l1 else None
            l2 = l2.next if l2 else None
        
        return res.next




        